readthedocs/projects/models.py

```diff
@@ -898,3 +905,36 @@ class WebHook(Notification):
 
     def __unicode__(self):
         return self.url
+
+
+class Domain(models.Model):
+    project = models.ForeignKey(Project, related_name='domains')
+    objects = RelatedProjectManager()
+
+    url = models.URLField(_('URL'), unique=True)
+
+    machine = models.BooleanField(
+        default=False, help_text=_('This URL was auto-created')
+    )
+    cname = models.BooleanField(
+        default=False, help_text=_('This URL is a CNAME for the project')
+    )
+    canonical = models.BooleanField(
+        default=False, help_text=_('This URL is the primary one where the documentation is served from.')
+    )
```

Comment:
It's a good practice to prefix boolean attributes/field names with `is_`, e.g. `is_canonical` which makes it clear what to expect from field and is better for reading expressions like `domain.is_cname`.

We don't follow that practice in the rest of the project, but maybe here is a good place to start with it?